arbeitszeit/use_cases/register_accountant.py

```diff
@@ -13,13 +20,34 @@ class Request:
         token: str
         email: str
         name: str
+        password: str
 
     @dataclass
     class Response:
         is_accepted: bool
+        user_id: Optional[UUID]
 
     token_service: InvitationTokenValidator
+    member_repository: MemberRepository
+    company_repository: CompanyRepository
+    accountant_repository: AccountantRepository
 
     def register_accountant(self, request: Request) -> Response:
         invited_email = self.token_service.unwrap_invitation_token(request.token)
-        return self.Response(is_accepted=invited_email == request.email)
+        if invited_email != request.email:
+            return self._failed_registration()
+        if self.member_repository.has_member_with_email(request.email):
```

Comment:
this makes that an accountant cannot use the same mail adress he/she uses for their company or member account. is this restriction necessary?